### PR TITLE
[SDL-0261] New vehicle data WindowStatus

### LINKED ISSUE #1508: [SDL 0261] New vehicle data WindowStatus
Proposal: [New vehicle data WindowStatus](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0261-New-vehicle-data-WindowStatus.md)

> This proposal is to add new vehicle data `WindowStatus` to know the position of windows in a vehicle.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/869

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the revision to `WindowStatus` described in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/869#issuecomment-564050172).

> The proposal .md file was updated to reflect these revisions on 12/12/19.

### PR DESCRIPTION
**Pull request by yoooriii**
Fixes #1508 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR

#### Unit Tests
Tests implemented and ran on my local machine on iPhone-8 simulator running iOS-13.0.

#### Core Tests
None

### Summary
[detailed description on sdl evolution](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0261-New-vehicle-data-WindowStatus.md)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by yoooriii** [comment]
@atiwari9 @YarikMamykin would you please review it?

File: SmartDeviceLink-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1543,6 +1543,12 @@
...
 		9FE2471922D77AED00F8D2FC /* SDLPredefinedWindows.h in Headers */ = {isa = PBXBuildFile; fileRef = 9FE2471722D77AED00F8D2FC /* SDLPredefinedWindows.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		B38D8E6524A029F700B977D0 /* SDLWindowStatus.m in Sources */ = {isa = PBXBuildFile; fileRef = B38D8E6324A029F700B977D0 /* SDLWindowStatus.m */; };
+		B38D8E6624A029F700B977D0 /* SDLWindowStatus.h in Headers */ = {isa = PBXBuildFile; fileRef = B38D8E6424A029F700B977D0 /* SDLWindowStatus.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		B38D8E6924A02A4800B977D0 /* SDLWindowState.h in Headers */ = {isa = PBXBuildFile; fileRef = B38D8E6724A02A4700B977D0 /* SDLWindowState.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		B38D8E6A24A02A4800B977D0 /* SDLWindowState.m in Sources */ = {isa = PBXBuildFile; fileRef = B38D8E6824A02A4800B977D0 /* SDLWindowState.m */; };
+		B38D8E7B24A0E11C00B977D0 /* SDLWindowStatusSpec.m in Sources */ = {isa = PBXBuildFile; fileRef = B38D8E7924A0E11C00B977D0 /* SDLWindowStatusSpec.m */; };
+		B38D8E7C24A0E11C00B977D0 /* SDLWindowStateSpec.m in Sources */ = {isa = PBXBuildFile; fileRef = B38D8E7A24A0E11C00B977D0 /* SDLWindowStateSpec.m */; };
```
Comment (SatbirTanda) [inline SmartDeviceLink-iOS.xcodeproj/project.pbxproj:1551]: Add new files to the `podspec` files

File: SmartDeviceLink/SDLGetVehicleDataResponse.h [hunk 2]
```diff
@@ -1,37 +1,62 @@
...
-@class SDLGPSData;
+@class SDLClusterModeStatus;
 @class SDLHeadLampStatus;
+@class SDLDeviceStatus;
+@class SDLBeltStatus;
 @class SDLMyKey;
+@class SDLGPSData;
 @class SDLTireStatus;
+@class SDLECallInfo;
+@class SDLAirbagStatus;
```
Comment (SatbirTanda) [inline SmartDeviceLink/SDLGetVehicleDataResponse.h:53]: Please try to keep imports in alphabetical order

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m [hunk 3]
```diff
@@ -44,6 +44,7 @@
         myKey = [[SDLMyKey alloc] init];
         fuelRange = [[SDLFuelRange alloc] init];
         cloudAppVehicleID = @"testCloudAppVehicleID";
+        windowStatus = @[[SDLWindowStatus new], [SDLWindowStatus new], [SDLWindowStatus new]];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m:47]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m [hunk 4]
```diff
@@ -144,7 +147,9 @@
                                            SDLRPCParameterNameTirePressure:tires,
                                            SDLRPCParameterNameTurnSignal:SDLTurnSignalOff,
                                            SDLRPCParameterNameVIN:@"222222222722",
-                                           SDLRPCParameterNameWiperStatus:SDLWiperStatusStalled},
+                                           SDLRPCParameterNameWiperStatus:SDLWiperStatusStalled,
+                                           @"windowStatus":windowStatus
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m:151]: Use `SDLWindowStateWindowStatus`

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m [hunk 5]
```diff
@@ -181,10 +186,11 @@
         expect(testNotification.turnSignal).to(equal(SDLTurnSignalOff));
         expect(testNotification.vin).to(equal(@"222222222722"));
         expect(testNotification.wiperStatus).to(equal(SDLWiperStatusStalled));
+        expect(testNotification.windowStatus).to(equal(windowStatus));
     });
     
     it(@"Should return nil if not set", ^ {
-        SDLOnVehicleData* testNotification = [[SDLOnVehicleData alloc] init];
+        SDLOnVehicleData* testNotification = [SDLOnVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m:193]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m [hunk 6]
```diff
@@ -216,15 +222,15 @@
         expect(testNotification.turnSignal).to(beNil());
         expect(testNotification.vin).to(beNil());
         expect(testNotification.wiperStatus).to(beNil());
+        expect(testNotification.windowStatus).to(beNil());
     });
 
     it(@"Should set and get generic Network Signal Data", ^{
-        SDLOnVehicleData *testRequest = [[SDLOnVehicleData alloc] init];
-
+        SDLOnVehicleData *testRequest = [SDLOnVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m:229]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m [hunk 7]
```diff
@@ -110,7 +112,9 @@
                                                                    SDLRPCParameterNameSteeringWheelAngle:@NO,
                                                                    SDLRPCParameterNameTirePressure:@YES,
                                                                    SDLRPCParameterNameTurnSignal:@NO,
-                                                                   SDLRPCParameterNameWiperStatus:@YES},
+                                                                   SDLRPCParameterNameWiperStatus:@YES,
+                                                                   @"windowStatus":@YES
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:116]: Use `SDLRPCParameterNameWindowStatus`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m [hunk 8]
```diff
@@ -146,12 +150,13 @@
...
         expect(testRequest.wiperStatus).to(equal(@YES));
+        expect(testRequest.windowStatus).to(equal(@YES));
     });
 });
 
 describe(@"initializers", ^{
-    context(@"init", ^{
-        SDLGetVehicleData* testRequest = [[SDLGetVehicleData alloc] init];
+    it(@"init", ^{
+        SDLGetVehicleData* testRequest = [SDLGetVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:159]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m [hunk 9]
```diff
@@ -182,126 +187,272 @@
...
+             expect(testRequest.myKey).to(beTrue());
+             expect(testRequest.odometer).to(beTrue());
+             expect(testRequest.prndl).to(beTrue());
+             expect(testRequest.rpm).to(beTrue());
+             expect(testRequest.speed).to(beTrue());
+             expect(testRequest.steeringWheelAngle).to(beTrue());
+             expect(testRequest.tirePressure).to(beTrue());
+             expect(testRequest.turnSignal).to(beTrue());
+             expect(testRequest.wiperStatus).to(beTrue());
+             expect(testRequest.windowStatus).to(beTrue());
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:374]: Stick to `.to(equal(@YES));`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m [hunk 10]
```diff
@@ -182,126 +187,272 @@
...
+                      expect(testRequest.myKey).to(beFalse());
+                      expect(testRequest.odometer).to(beFalse());
+                      expect(testRequest.prndl).to(beFalse());
+                      expect(testRequest.rpm).to(beFalse());
+                      expect(testRequest.speed).to(beFalse());
+                      expect(testRequest.steeringWheelAngle).to(beFalse());
+                      expect(testRequest.tirePressure).to(beFalse());
+                      expect(testRequest.turnSignal).to(beFalse());
+                      expect(testRequest.wiperStatus).to(beFalse());
+                      expect(testRequest.windowStatus).to(beFalse());
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:441]: .to(equal(@NO));

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 11]
```diff
@@ -16,7 +16,7 @@
 
 describe(@"Getter/Setter Tests", ^ {
     it(@"Should set and get correctly", ^ {
-        SDLSubscribeVehicleData* testRequest = [[SDLSubscribeVehicleData alloc] init];
+        SDLSubscribeVehicleData* testRequest = [SDLSubscribeVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:19]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 12]
```diff
@@ -77,6 +78,7 @@
         expect(testRequest.tirePressure).to(equal(@NO));
         expect(testRequest.turnSignal).to(equal(@YES));
         expect(testRequest.wiperStatus).to(equal(@NO));
+        expect(testRequest.windowStatus).to(beTrue());
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:81]: Stick to `.to(equal(@YES));`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 13]
```diff
@@ -110,7 +112,9 @@
                                                            SDLRPCParameterNameSteeringWheelAngle:@NO,
                                                            SDLRPCParameterNameTirePressure:@YES,
                                                            SDLRPCParameterNameTurnSignal:@NO,
-                                                           SDLRPCParameterNameWiperStatus:@NO},
+                                                           SDLRPCParameterNameWiperStatus:@NO,
+                                                           @"windowStatus":@YES
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:116]: Use `SDLRPCParameterNameWindowStatu`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 14]
```diff
@@ -146,12 +150,13 @@
...
         expect(testRequest.wiperStatus).to(equal(@NO));
+        expect(testRequest.windowStatus).to(equal(@YES));
     });
 });
 
 describe(@"initializers", ^{
-    context(@"init", ^{
-        SDLSubscribeVehicleData* testRequest = [[SDLSubscribeVehicleData alloc] init];
+    it(@"init", ^{
+        SDLSubscribeVehicleData* testRequest = [SDLSubscribeVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:159]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 15]
```diff
@@ -182,10 +187,11 @@
...
         expect(testRequest.turnSignal).to(beNil());
         expect(testRequest.wiperStatus).to(beNil());
+        expect(testRequest.windowStatus).to(beNil());
     });
 
-    context(@"initWithAccelerationPedalPosition:airbagStatus:beltStatus:bodyInformation:cloudAppVehicleID:clusterModeStatus:deviceStatus:driverBraking:eCallInfo:emergencyEvent:engineOilLife:engineTorque:externalTemperature:fuelLevel:fuelLevelState:gps:headLampStatus:instantFuelConsumption:myKey:odometer:prndl:rpm:speed:steeringWheelAngle:tirePressure:wiperStatus:", ^{
-        SDLSubscribeVehicleData* testRequest = [[SDLSubscribeVehicleData alloc] initWithAccelerationPedalPosition:YES airbagStatus:YES beltStatus:YES bodyInformation:YES cloudAppVehicleID:YES clusterModeStatus:YES deviceStatus:YES driverBraking:YES eCallInfo:YES electronicParkBrakeStatus:YES emergencyEvent:YES engineOilLife:YES engineTorque:YES externalTemperature:YES fuelLevel:YES fuelLevelState:YES fuelRange:YES gps:YES headLampStatus:YES instantFuelConsumption:YES myKey:YES odometer:YES prndl:YES rpm:YES speed:YES steeringWheelAngle:YES tirePressure:YES turnSignal:YES wiperStatus:YES];
+    it(@"INIT-4", ^{
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:193]: Stick to `@"initWithAccelerationPedalPosition:airbagStatus:beltStatus:bodyInformation:cloudAppVehicleID:clusterModeStatus:deviceStatus:driverBraking:eCallInfo:emergencyEvent:engineOilLife:engineTorque:externalTemperature:fuelLevel:fuelLevelState:gps:headLampStatus:instantFuelConsumption:myKey:odometer:prndl:rpm:speed:steeringWheelAngle:tirePressure:wiperStatus:"`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m [hunk 16]
```diff
@@ -16,7 +16,7 @@
 
 describe(@"Getter/Setter Tests", ^ {
     it(@"Should set and get correctly", ^ {
-        SDLUnsubscribeVehicleData* testRequest = [[SDLUnsubscribeVehicleData alloc] init];
+        SDLUnsubscribeVehicleData* testRequest = [SDLUnsubscribeVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:19]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m [hunk 17]
```diff
@@ -110,7 +112,9 @@
                                                            SDLRPCParameterNameSteeringWheelAngle:@YES,
                                                            SDLRPCParameterNameTirePressure:@YES,
                                                            SDLRPCParameterNameTurnSignal:@YES,
-                                                           SDLRPCParameterNameWiperStatus:@YES},
+                                                           SDLRPCParameterNameWiperStatus:@YES,
+                                                           @"windowStatus":@YES
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:116]: Use `SDLRPCParameterNameWindowStatus `

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m [hunk 18]
```diff
@@ -146,12 +150,13 @@
...
         expect(testRequest.wiperStatus).to(equal(@YES));
+        expect(testRequest.windowStatus).to(equal(@YES));
     });
 });
 
 describe(@"initializers", ^{
-    context(@"init", ^{
-        SDLUnsubscribeVehicleData *testRequest = [[SDLUnsubscribeVehicleData alloc] init];
+    it(@"init", ^{
+        SDLUnsubscribeVehicleData *testRequest = [SDLUnsubscribeVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:159]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m [hunk 19]
```diff
@@ -182,122 +187,132 @@
...
+            expect(testRequest.turnSignal).to(equal(@NO));
+            expect(testRequest.wiperStatus).to(equal(@YES));
+            expect(testRequest.windowStatus).to(equal(@NO));
+        });
     });
 
     context(@"Should set and get Generic Network Signal Data", ^{
-        SDLUnsubscribeVehicleData *testRequest = [[SDLUnsubscribeVehicleData alloc] init];
-
+        SDLUnsubscribeVehicleData *testRequest = [SDLUnsubscribeVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:307]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m [hunk 20]
```diff
@@ -117,7 +118,9 @@
                                                                    SDLRPCParameterNameSteeringWheelAngle:vehicleDataResult,
                                                                    SDLRPCParameterNameTirePressure:vehicleDataResult,
                                                                    SDLRPCParameterNameTurnSignal:vehicleDataResult,
-                                                                   SDLRPCParameterNameWiperStatus:vehicleDataResult},
+                                                                   SDLRPCParameterNameWiperStatus:vehicleDataResult,
+                                                                   @"windowStatus":vehicleDataResult
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m:122]: Use `SDLRPCParameterNameWindowStatus`

File: SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m [hunk 21]
```diff
@@ -153,10 +156,11 @@
         expect(testResponse.tirePressure).to(equal(vehicleDataResult));
         expect(testResponse.turnSignal).to(equal(vehicleDataResult));
         expect(testResponse.wiperStatus).to(equal(vehicleDataResult));
+        expect(testResponse.windowStatus).to(equal(vehicleDataResult));
     });
     
     it(@"Should return nil if not set", ^ {
-        SDLSubscribeVehicleDataResponse* testResponse = [[SDLSubscribeVehicleDataResponse alloc] init];
+        SDLSubscribeVehicleDataResponse* testResponse = [SDLSubscribeVehicleDataResponse new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m:163]: Use `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLUnsubscribeVehicleDataResponseSpec.m [hunk 22]
```diff
@@ -152,10 +155,11 @@
         expect(testResponse.electronicParkBrakeStatus).to(equal(vehicleDataResult));
         expect(testResponse.turnSignal).to(equal(vehicleDataResult));
         expect(testResponse.cloudAppVehicleID).to(equal(vehicleDataResult));
+        expect(testResponse.windowStatus).to(equal(vehicleDataResult));
     });
     
     it(@"Should return nil if not set", ^ {
-        SDLUnsubscribeVehicleDataResponse* testResponse = [[SDLUnsubscribeVehicleDataResponse alloc] init];
+        SDLUnsubscribeVehicleDataResponse* testResponse = [SDLUnsubscribeVehicleDataResponse new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLUnsubscribeVehicleDataResponseSpec.m:162]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowStateSpec.m [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+        SDLWindowState *testStruct = [[SDLWindowState alloc] initWithApproximatePosition:approximatePosition deviation:deviation];
+
+        expect(testStruct.approximatePosition).to(equal(approximatePosition));
+        expect(testStruct.deviation).to(equal(deviation));
+    });
+
+    it(@"Should get correctly when initialized", ^ {
+        NSDictionary *dict = @{@"deviation":@(deviation),
+                                @"approximatePosition":@(approximatePosition)
+                                };
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowStateSpec.m:27]: Use `SDLRPCParameterNameDeviation` and `SDLRPCParameterNameApproximatePosition`

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowStateSpec.m [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+        SDLWindowState *testStruct = [[SDLWindowState alloc] initWithDictionary:dict];
+#pragma clang diagnostic pop
+
+        expect(testStruct.approximatePosition).to(equal(approximatePosition));
+        expect(testStruct.deviation).to(equal(deviation));
+    });
+
+    it(@"Should return nil if not set", ^ {
+        SDLWindowState *testStruct = [SDLWindowState new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowStateSpec.m:38]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowStatusSpec.m [hunk 25]
```diff
@@ -0,0 +1,51 @@
...
+        SDLWindowStatus *testStruct = [[SDLWindowStatus alloc] initWithLocation:location state:state];
+
+        expect(testStruct.location).to(equal(location));
+        expect(testStruct.state).to(equal(state));
+    });
+
+    it(@"Should get correctly when initialized", ^ {
+        SDLWindowState *state = [[SDLWindowState alloc] initWithApproximatePosition:approximatePosition deviation:deviation];
+        SDLGrid *location = [SDLGrid new];
+        NSDictionary *dict = @{@"location":location,
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowStatusSpec.m:31]: Use the SDLRPCParameterNames

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowStatusSpec.m [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+        SDLWindowStatus *testStruct = [[SDLWindowStatus alloc] initWithDictionary:dict];
+#pragma clang diagnostic pop
+
+        expect(testStruct.location).to(equal(location));
+        expect(testStruct.state).to(equal(state));
+    });
+
+    it(@"Should return nil if not set", ^ {
+        SDLWindowStatus *testStruct = [SDLWindowStatus new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLWindowStatusSpec.m:44]: `alloc init`

**Comment by yoooriii** [comment]
Hello, @theresalech could you please review?

**Comment by theresalech** [comment]
@yoooriii please make sure the feedback provided in https://github.com/smartdevicelink/sdl_ios/pull/1693#pullrequestreview-450716620 is reflected in this PR as well, and let us know once this has been confirmed/ready for review.  Thank you!

**Comment by yoooriii** [comment]
Hello @theresalech. I have fixed this PR following the #1693 review comments. Could you please review it?

File: SmartDeviceLink/SDLOnVehicleData.h [hunk 27]
```diff
@@ -22,169 +50,272 @@
...
- Since SmartDeviceLink 2.0
+ * Callback for the periodic and non periodic vehicle data read function.
+ *
+ * @since SDL 2.0.0
  */
+@interface SDLOnVehicleData : SDLRPCNotification
 
-NS_ASSUME_NONNULL_BEGIN
+/**
+ * note: RPC generator produced initializer
```
Comment (NicoleYarroch) [inline SmartDeviceLink/SDLOnVehicleData.h:65]:
```suggestion
 *  Convenience init for setting all possible on vehicle data items.
```

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 28]
```diff
@@ -4,13 +4,12 @@
 
 
 #import <Foundation/Foundation.h>
-
-#import <Quick/Quick.h>
 #import <Nimble/Nimble.h>
+#import <Quick/Quick.h>
```
Comment (NicoleYarroch) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:8]: Tests are failing in this class.

Comment (yoooriii) [reply SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:7]: fixed

**Comment by yoooriii** [comment]
The 2 fallen tests are: SDLAudioStreamManagerSpec, SDLLockScreenManagerSpec. Neither one is connected with this pull request, besides I cannot reproduce it.

**Comment by joeljfischer** [comment]
@yoooriii, we will not be reviewing this PR again until [this PR](https://github.com/smartdevicelink/sdl_ios/pull/1693) is fully reviewed and merged in order to reduce the amount of time that we are spending in giving the same review comments and so that you have an approved PR to know exactly what we're looking for. We ask that you place your focus on that PR.

**Comment by yoooriii** [comment]
Hello @joeljfischer, could you please review the latest changes? Since the referenced PR is already merged this PR has merge conflicts. Should I merge it with develop branch in order to pass this review?

**Comment by joeljfischer** [comment]
@yoooriii hello, yes merge conflicts must be resolved for the review to be approved. If you only want to do one at a time and have me review that one so that you’re not updating a bunch at once, that is fine as well

**Comment by yoooriii** [comment]
@joeljfischer Ok it is clear, since merge conflicts inevitable let us do it one by one, let #1692  be next and then I go on with this one.

**Comment by yoooriii** [comment]
Hello @joeljfischer, could you please take a look at this PR? I have resolved merge conflicts. (As always the only fallen test is SDLLockScreenManagerSpec, it has nothing to do with the PR)

**Comment by yoooriii** [comment]
> @yoooriii It looks like code from the ["Gear Status" PR](https://github.com/smartdevicelink/sdl_ios/pull/1690) accidentally got merged into this PR. Can you remove the code related to the "Gear Status" PR and let me know when this PR is ready for re-review? Thanks.

I am sorry, it looks like that. In such case let us wait until #1690 merged with develop, after that it must be fine to use it as it is now.

**Comment by NicoleYarroch** [comment]
@yoooriii The Gear Status PR #1690 was approved and merged. I can see that there are now a few files that have merge conflicts with the files on the develop branch. Can you fix the merge conflicts and tag me when this PR is ready for review. Thanks.

**Comment by theresalech** [comment]
Hi @yoooriii, can you please advise when this PR will be ready for re-review? As we are nearing the end of our development timeline for this release, we want to make sure all PRs are addressed as soon as possible. Please let us know if you have any questions about our latest review. Thank you!

**Comment by yoooriii** [comment]
Hello @NicoleYarroch , @theresalech , the PR is ready, tests passed successfully (the errors are connected with: Upload coverage to Codecov)

**Comment by jordynmackool** [comment]
@yoooriii is this ready for review?

**Comment by yoooriii** [comment]
Hello. @jordynmackool yes it is, @NicoleYarroch could you please review?

**Pull request merged**